src/directory-sync/utils/get-primary-email.ts

```diff
@@ -0,0 +1,6 @@
+import { User } from '../interfaces/user.interface';
+
+export function getPrimaryEmail(user: User): string | undefined {
+  const primaryEmail = (user.emails || []).find((email) => email.primary);
```

Comment:
We should be able to use optional chaining here instead of adding a default value:

```suggestion
  const primaryEmail = user.emails?.find((email) => email.primary);
```